device/mellanox/x86_64-mlnx_msn2700-r0/plugins/fanutil.py

```
@@ -0,0 +1,198 @@
+        list_of_fan_leds = glob(self.fan_path + self.fan_led_green)
+        num_of_drawer = len(list_of_fan_leds)
+
+        return num_of_fan, num_of_drawer
+
+    def _convert_fan_index_to_drawer_index(self, index):
+        return (index + 1) / 2
+
+    def _read_file(self, file_pattern, index = 0):
+        """
+        Reads the file of the fan
+
+        :param file_pattern: The filename convention
+        :param index: An integer, 1-based index of the fan of which to query status
+        :return: int
+        """
+        return_value = 0
+        try:
+            with open(self.fan_path + file_pattern.format(index), 'r') as file_to_read:
+                return_value = int(file_to_read.read())
+        except IOError:
+            log_err("Read file {} failed".format(self.fan_path + file_pattern.format(index)))
+            return 0
+
+        return return_value
+
+    def get_num_fans(self):
+        """
+        Retrieves the number of FANs supported on the device
+
+        :return: An integer, the number of FANs supported on the device
+        """
+        return self.num_of_fan
+
+    def get_status(self, index):
+        """
+        Retrieves the operational status of FAN defined
+                by index 1-based <index>
+
+        :param index: An integer, 1-based index of the PSU of which to query status
+        :return: Boolean,
+            - True if FAN is running with some speed 
+            - False if FAN has stopped running
+        """
+        if not self.get_presence(index):
+            return False
+
+        return self.get_speed(index) != 0
+
+    def get_presence(self, index):
+        """
+        Retrieves the presence status of a FAN defined
+                by 1-based index <index>
+
+        :param index: An integer, 1-based index of the FAN of which to query status
+        :return: Boolean, True if FAN is plugged, False if not
+        """
+        if index > self.num_of_fan:
+            return False
```

Consider raise an exception instead of return False.

---

fixed.